### DISCUSSION TITLE
Reporter System Vector of Tuples

### DISCUSSION THREAD

**Discussion by maxnezdyur** [idaholab/moose#23734] [Q&A General] [answered]
I want to be able to send a vector of line segments from one multi-app to another. Is it possible to send a vector of tuples that hold the 6 coordinate values with the reporter system? I know I can send 6 reporters of each coord individually but I want to make the code cleaner and easier to read.  Let me know what would need to be done or if there was an easier way to send over a vector of line segments.

**Comment by GiudGiud** [comment]
The reporter is the right data type to look at. I m not sure which one of vector of tuple or vector of vector is supported natively

@loganharbour or @zachmprince may advise you further.

**Comment by loganharbour** [comment]
Reporters just require a `dataStore` and `dataLoad` specialization for the object, plus a `to_json` specialization for converting to JSON. 

We already have `Point` specialized for `dataStore` and `dataLoad` in `DataIO.C`. You will probably need to specialize `to_json` for `Point` in `JsonIO.h`. I think our library should work with the rest.

I would just give it a try and to bother me with any compiler errors on Slack. Please put up a PR for whatever you need to get this working once it's all complete.

**Reply by loganharbour** [reply]
Update... I think we'll also need to specialize a tuple

**Reply by maxnezdyur** [reply]
It seems like if `Point` already has the `dataStore` and `dataLoad` specs, I could create two more for `LineSegment` that use the `Point` store/load and a `to_json` for both `Point` and `LineSegement`, then I would just have a vector of `LineSegments`.

**Reply by loganharbour** [reply]
Yep - that should be plenty sufficient. Specialize all of the above in the location where `LineSegment` is declared and defined.

**Reply by maxnezdyur** [reply]
For the `dataStore` and `dataLoad` for `LineSegment` should that be in `DataIO`?

**Reply by loganharbour** [reply, marked as answer]
Nope - the convention should be that if it's not a class in MOOSE, it should be specialized there. Otherwise, it should be specialized where the object itself is declared and defined.

For example: std library things, libmesh objects, etc in DataIO. Everything else outside of DataIO
